### PR TITLE
Fix group form errors

### LINKED ISSUE #191: Group Creation form errors not displaying
## Problem

Errors are not displaying on the Group creation form when validation fails for group description, name that has already been taken but path changed. When the group description is too long the page just redisplays, however creating a group with a name that already exists but with the path modified results in a `PG::UniqueViolation: ERROR: duplicate key value violates unique constraint "index_routes_on_name"`

## Steps to reproduce
1. From the Groups Dashboard click `New group`
2. Fill in `Group Name`, `Path` with `aaaaa`, and `Description` with a string longer than 255 characters
3. Click `Create group`
4. See that the form is re-rendered and no fields are show as in error and no errors are displayed. Verify in the server console that there was a `Rollback`
5. Now update the `Group Name` to `Staphylococcus` and `Path` to `staphylococcus1`
6. You should see an error page `ActiveRecord::RecordNotUnique in GroupsController#create`
7. Navigate back to Groups Dashboard and confirm that the group is not created as expected.

## Solution
Update Group Form to display errors when Group is not created due to failing validation

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Fixed displaying errors on create & edit group forms when validation fails for description & name that has already been taken. Sidebar & breadcrumbs are out of scope.
Fixes #191.

## Screenshots or screen recordings
Screenshot from 2023-09-19 09-54-16
Screenshot from 2023-09-15 15-04-37
Screenshot from 2023-09-15 15-10-24
Screenshot from 2023-09-15 15-03-44

## How to set up and validate locally
A. Test create group form validation.
   1. From the Groups Dashboard click `New group`
   2. Fill in `Group Name` with `a` and `Description` with a string longer than 255 characters.
   3. Click `Create group`. See that the form displays errors.
   4. Now update the `Group Name` to `Staphylococcus`.
   5. Click `Create group`. See that the form displays errors.
   6. Navigate back to Groups Dashboard and confirm that the group is not created as expected.
   
B. Test edit group form validation.
   1. From the Groups Dashboard click the `Staphylococcus` group.
   2. Fill in `Group Name` with `a` and `Description` with a string longer than 255 characters.
   3. Click `Save Changes`. See that the form displays errors.
   4. Navigate back to Groups Dashboard and confirm that the group was not updated as expected.
   
Note: Repeat tests for subgroups.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/system/groups_test.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
```
Comment (deepsidhu85) [inline test/system/groups_test.rb:2]: Can you please add a test which tests the following condition:

1) A `Name` that isn't duplicate of an existing group and `Path` that already exists for a group

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good! Just the one comment below for testing

File: test/system/groups_test.rb (hunk 1 above)
Comment (ksierks) [reply test/system/groups_test.rb:2]: Please see cc8bd1e7e8d81a609bedee8e118d728a65e5882e.

Comment (deepsidhu85) [reply test/system/groups_test.rb:2]: Awesome thanks!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**